### PR TITLE
feat: nested models in spec

### PR DESCRIPTION
**Pull request by TorstenDittmann**
## What does this PR do?

- adds a recursive function to get all nested models - previously we only went "one level" deep
- this allows us to finally generate Console SDK without issues

## Test Plan

- open api & swagger specs are more valid now

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?

✅

### CODE REVIEW

File: src/Appwrite/Specification/Format.php [hunk 1]
```diff
@@ -8,40 +8,22 @@
...
-     */
-    protected $authCount;
-
-    /**
-     * @var array
-     */
-    protected $params = [
+    protected array $services;
+    protected array $keys;
+    protected int $authCount;
```
Comment (lohanidamodar) [inline src/Appwrite/Specification/Format.php:25]: Doc comment missing

File: src/Appwrite/Specification/Format.php [hunk 2]
```diff
@@ -56,14 +38,6 @@ abstract class Format
...
 
-    /**
-     * @param App $app
-     * @param array $services
-     * @param Route[] $routes
-     * @param Model[] $models
-     * @param array $keys
-     * @param int $authCount
-     */
     public function __construct(App $app, array $services, array $routes, array $models, array $keys, int $authCount)
```
Comment (lohanidamodar) [inline src/Appwrite/Specification/Format.php:41]: Did we decide to remove doc comments in favor of type definition?

**Review by lohanidamodar — APPROVED** [review]
LGTM, just a query

File: src/Appwrite/Specification/Format.php (hunk 2 above)
Comment (TorstenDittmann) [reply src/Appwrite/Specification/Format.php:41]: I don't see too much value for doc types anymore when not needed.

It just creates a lot of noise in the code and increases the code we have to write, without giving any value.

File: app/tasks/specs.php [hunk 3]
```diff
@@ -123,105 +123,94 @@
...
+        foreach ($platforms as $platform) {
+            $routes = [];
+            $models = [];
+            $services = [];
+
+            foreach ($appRoutes as $key => $method) {
+                foreach ($method as $route) {
+                    /** @var \Utopia\Route $route */
+                    $routeSecurity = $route->getLabel('sdk.auth', []);
+                    $sdkPlatofrms = [];
```
Comment (abnegate) [inline app/tasks/specs.php:135]: Since you have fixed them elsewhere, could we fix this name typo too

**Review by abnegate — APPROVED** [review]
Looks good!

**Pull request merged**